src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -0,0 +1,611 @@
+/*-------------------------------------------------------------------------
+ *
+ * remote_transaction.c
+ *   Management of transaction spanning more than one node.
+ *
+ * Copyright (c) 2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "libpq-fe.h"
+
+#include "miscadmin.h"
+
+#include "access/xact.h"
+#include "distributed/connection_management.h"
+#include "distributed/transaction_management.h"
+#include "distributed/remote_commands.h"
+#include "distributed/remote_transaction.h"
+#include "utils/hsearch.h"
+
+
+static void CheckTransactionHealth(void);
+static void Assign2PCIdentifier(MultiConnection *connection);
+static void WarnAboutLeakedPreparedTransaction(MultiConnection *connection, bool commit);
+
+
+/*
+ * AdjustRemoteTransactionStates Begins, if necessary according to this
+ * session's coordinated transaction state, and the remote transaction's
+ * state, an explicit transaction on all the connections.  This is done in
+ * parallel, to lessen latency penalties.
+ */
+void
+AdjustRemoteTransactionStates(List *connectionList)
```

Comment:
Still would like this to be something like `InitializeRemoteTransaction`. Adjust.. feels like we're just fixing up a few variables (like `AdjustColumnOldAttributes`, `AdjustStateForFailure`).